### ISSUE TITLE
Return W3002 instead of E3002 for additional resources supported by package command

### ISSUE THREAD

**Issue by asjohnston-asf** [aws-cloudformation/cfn-python-lint#1683]
cfn-lint version: `cfn-lint 0.35.1`

Some resources supported by `aws cloudformation package` return `E3002 Expecting an object...`; they should instead return `W3002 This code may only work with package cli command...` when the resource value is a string.

Example:
```
$ cfn-lint --version
cfn-lint 0.35.1

$ cat myTemplate.yml 
Resources:

  myStateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: myRoleArn
      DefinitionS3Location: myLocalDefinitionFile.json

$ cfn-lint myTemplate.yml 
E3002 Expecting an object at Resources/myStateMachine/Properties/DefinitionS3Location
myTemplate.yml:7:7
```

My team is specifically interested in support for [`AWS::StepFunctions:StateMachine DefinitionS3Location`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-stepfunctions-statemachine.html#cfn-stepfunctions-statemachine-definitions3location).  We would prefer to use `package` for our step function definitions so we can perform static analysis on the definition separate from the cloudformation template.  That would currently require an `--ignore-checks E3002` flag when validating the template, but we consider that check too critical to ignore.

Resources currently supported by `package` are documented at:
https://docs.aws.amazon.com/cli/latest/reference/cloudformation/package.html

Exceptions for some of these resources are implemented in `cfn-lint` at:
https://github.com/aws-cloudformation/cfn-python-lint/blob/fe0af928a4ab26a6cedfd76564a26c9a4120b14f/src/cfnlint/rules/resources/properties/Properties.py#L179-L184
https://github.com/aws-cloudformation/cfn-python-lint/blob/fe0af928a4ab26a6cedfd76564a26c9a4120b14f/src/cfnlint/rules/resources/properties/PropertiesTemplated.py#L42-L47

**Comment by asjohnston-asf** [comment]
Adding an exception for DefinitionS3Location ito resolve the issue with my example is simple enough: https://github.com/aws-cloudformation/cfn-python-lint/compare/master...asjohnston-asf:master

However, I haven't found any test coverage for these exceptions, I don't appreciate any nuance of the property type being `S3Location` vs `DefinitionS3Location`, and it appears to work with/without the added item in `self.resource_property_types.extend`, so I'm inclined to leave the PR to the professionals.

**Comment by PatMyron** [comment]
Thanks for the report. Should be fixed for the next upcoming release

Can workaround with [resource-level ignores](https://github.com/aws-cloudformation/cfn-python-lint#resource-based-metadata) until then:

```yaml
  Resource:
    Type: AWS::StepFunctions::StateMachine
    Metadata:
      cfn-lint:
        config:
          ignore_checks:
          - E3002
```

**Comment by asjohnston-asf** [comment]
Thanks for the quick turnaround!

**Comment by ozbillwang** [comment]
Could we reopen this ticket?

Seems still get the error, with release 0.44.7 and 0.46.0, still get below error. 

```
W3002 This code may only work with `package` cli command as the property
```

**Comment by stevehaneytrailblazer** [comment]
I agree on this one as status code 4 is treated as an error in our GitAction failing our CI/CD

**Comment by isuftin** [comment]
@stevehaneytrailblazer  - Seeing this today as well. 

`W3002     This code may only work with `package` cli command as the property (Resources/IAM/Properties/TemplateURL) is a string`

CFN snippet:

```yaml
  IAM:
    Type: AWS::CloudFormation::Stack
    Properties:
      TemplateURL: './iam.yml'
      Parameters:
        Environment: !Ref Environment
        OidcProvider:
          !Select [1, !Split ['//', !GetAtt EKS.Outputs.OpenIdConnectIssuerUrl]]
```

**Comment by stevehaneytrailblazer** [comment]
@isuftin  fwiw I found there are options to ignore warnings altogether or specific codes so you could employ one of these:

`cfn-lint -t cf-templates/global-resources/cd.yaml -i W`   which would ignore all warnings

`cfn-lint -t cf-templates/global-resources/cd.yaml -i W3002`  ignores just that specific code

**Comment by ozbillwang** [comment]
@stevehaneytrailblazer 

Of course , you can ignore. But this is not the right way. 

The problem here is, the definiation for this error , it is not **error**(**E**xxxx), it should be belong to **warning** (**W**xxxx)

Hope you get the point.

E3002 to W3002
